### PR TITLE
Added e2e tests for Node

### PR DESCRIPTION
**Pull request by mrickard**
This adds e2e tests for Node. This also adds the reusable test and publish workflows, which Node is using.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+on:
+  workflow_call:
+    inputs:
+      INITCONTAINER_LANGUAGE:
+        required: true
+        type: string
+      K8S_OPERATOR_IMAGE_TAG:
+        required: true
+        type: string
+      RUNTIME_VERSION:
```
Comment (tippmar-nr) [inline .github/workflows/test.yml:26]: I'm not seeing where this input is being used anywhere in this workflow.

Comment (mrickard) [reply .github/workflows/test.yml:26]: I've added a reference in test.yml to pass it to the appropriate step. In practice, I think only Node is going to be using that.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          INITCONTAINER_LANGUAGE: ${{ env.INITCONTAINER_LANGUAGE }}
+          K8S_OPERATOR_IMAGE_TAG: ${{ env.K8S_OPERATOR_IMAGE_TAG }}
+          RUNTIME_VERSION: ${{ matrix.node-version }}
+
+      - name: Get agent version
+        run: |
+          if [[ ${{ matrix.agent-version }} == 'latest' ]]; then
+            cd nodejs && npm i newrelic@${{ matrix.agent-version }} && cd ..
+            echo AGENT_VERSION=$(cat nodejs/package.json | jq -r '.dependencies."newrelic"') >> $GITHUB_ENV 
+          elif
```
Comment (tippmar-nr) [inline .github/workflows/nodejs.yml:71]: should this be `else`? (I'm no bash expert...)

**Review by tippmar-nr — APPROVED** [review]

File: .github/workflows/nodejs.yml (hunk 2 above)
Comment (mrickard) [reply .github/workflows/nodejs.yml:71]: `elif` is correct for bash.

**Review by tippmar-nr — APPROVED** [review]

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+env:
+  INITCONTAINER_LANGUAGE: nodejs
+  K8S_OPERATOR_IMAGE_TAG: edge
+
+
+jobs:
+  call-test:
+    uses: ./.github/workflows/test.yml
+    with:
```
Comment (tippmar-nr) [inline .github/workflows/nodejs.yml:44]: You'll need to add `secrets: inherit` here in order for `test.yml` to have access to Github secrets.

Comment (tippmar-nr) [reply .github/workflows/nodejs.yml:44]: And also on the call to `publish.yml`

**Review by TimPansino — APPROVED** [review]
🚀  Let's send this one out and iterate on it further

**Pull request merged**